src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -134,5 +127,329 @@ private BoundExpression BindInterpolatedString(InterpolatedStringExpressionSynta
+                }
+                else
+                {
+                    Debug.Assert(parameterNamesAndLocationsBuilder.Count == 0 || parameterNamesAndLocationsBuilder[0] == null);
+                    parameterNamesAndLocations = default;
+                    parameterNamesAndLocationsBuilder.Clear();
+                }
+
+                var call = MakeInvocationExpression(part.Syntax, implicitBuilderReceiver, methodName, arguments, diagnostics, names: parameterNamesAndLocations, searchExtensionMethodsIfNecessary: false);
+                builderAppendCalls.Add(call);
+
+                // PROTOTYPE(interp-string): Handle dynamic
+                Debug.Assert(call is BoundCall or { HasErrors: true });
+
+                if (call is BoundCall { Method: { ReturnType: var returnType } method })
+                {
+                    bool methodReturnsBool = returnType.SpecialType == SpecialType.System_Boolean;
+                    if (!methodReturnsBool && returnType.SpecialType != SpecialType.System_Void)
+                    {
+                        // Interpolated string builder method '{0}' is malformed. It does not return 'void' or 'bool'.
+                        diagnostics.Add(ErrorCode.ERR_InterpolatedStringBuilderMethodReturnMalformed, part.Syntax.Location, method);
+                    }
+                    else if (builderPatternExpectsBool == null)
+                    {
+                        builderPatternExpectsBool = methodReturnsBool;
+                    }
+                    else if (builderPatternExpectsBool != methodReturnsBool)
+                    {
+                        // Interpolated string builder method '{0}' has inconsistent return types. Expected to return '{0}'.
+                        var expected = builderPatternExpectsBool == true ? Compilation.GetSpecialType(SpecialType.System_Boolean) : Compilation.GetSpecialType(SpecialType.System_Object);
+                        diagnostics.Add(ErrorCode.ERR_InterpolatedStringBuilderMethodReturnInconsistent, part.Syntax.Location, method, expected);
+                    }
+                }
+            }
+
+            argumentsBuilder.Free();
+            parameterNamesAndLocationsBuilder.Free();
+            return (builderAppendCalls.ToImmutableAndFree(), builderPatternExpectsBool ?? false);
+        }
+
+        private BoundCall? BindCreateCall(SyntaxNode syntax, TypeSymbol builderType, ImmutableArray<BoundExpression> args, BindingDiagnosticBag diagnostics)
+        {
+            var typeExpression = new BoundTypeExpression(syntax, aliasOpt: null, boundContainingTypeOpt: null, TypeWithAnnotations.Create(builderType)) { WasCompilerGenerated = true };
+
+            var expression = MakeInvocationExpression(syntax, typeExpression, "Create", args, diagnostics);
+            if (expression is BoundCall call && call.Type.Equals(builderType, TypeCompareKind.AllIgnoreOptions))
+            {
+                return call;
```

Comment:
>call [](http://example.com/codeflow?start=23&length=4)

Is there a guarantee that the node is marked as compiler generated? Otherwise we should consider marking it here.

Reply:
There is. `MakeInvocationExpression` is specifically for compiler-synthesized calls and unconditionally marks the invocation it returns as compiler-generated.